ui/components/app/snaps/snap-install-warning/snap-install-warning.js

```diff
@@ -50,47 +33,79 @@ export default function SnapInstallWarning({
   snapName,
 }) {
   const t = useI18nContext();
-  const [checkboxState, dispatch] = useReducer(checkboxStateReducer, {});
-
-  const isAllChecked = warnings.every((warning) => checkboxState[warning.id]);
-
-  const onCheckboxClicked = useCallback((checkboxId) => {
-    dispatch({ type: 'check', checkboxId });
-  }, []);
+  const [userAgree, setUserAgree] = useState(false);
 
   const SnapInstallWarningFooter = () => {
     return (
-      <div className="snap-install-warning__footer">
+      <Box
+        display={Display.Flex}
+        justifyContent={JustifyContent.center}
+        alignItems={AlignItems.center}
+        width={BlockSize.Full}
+      >
         <Button
-          className="snap-install-warning__footer-button"
           type="primary"
-          disabled={!isAllChecked}
+          disabled={!userAgree}
           onClick={onSubmit}
+          width={BlockSize.Full}
+          size={ButtonSize.Lg}
         >
           {t('confirm')}
         </Button>
-      </div>
+      </Box>
     );
   };
 
+  function constructWarningElementString(permissionWarnings) {
```

Comment:
This seems a bit confusing, it isn't really returning a string.

Also how is this array of text + components rendered? 🤔

Reply:
You're right, it's `string` in the UI context, but not here in component's code. I changed it to `constructWarningElementComponentArray`.

I think React knows somehow to combine these elements with strings and render them properly, but I changed everything to be a `<span>` element to be consistent.

![Screenshot 2023-11-03 at 12 02 11](https://github.com/MetaMask/metamask-extension/assets/13301024/7206de76-16ef-40ec-9abb-03432616fb6a)